app/experimenter/experiments/tests/test_forms.py

```diff
@@ -354,7 +355,47 @@ def test_empty_branch_size_raises_validation_error(self):
         self.assertFalse(form.is_valid())
 
 
-class TestExperimentVariantsAddonForm(MockRequestMixin, TestCase):
+VARIANTS_FORM_FIXTURE_DATA = {
+    "population_percent": "10",
+    "firefox_version": Experiment.VERSION_CHOICES[-1][0],
+    "firefox_channel": Experiment.CHANNEL_NIGHTLY,
+    "client_matching": "en-us only please",
+    "locales": [],
+    "countries": [],
+    "pref_key": "browser.test.example",
+    "pref_type": Experiment.PREF_TYPE_STR,
+    "pref_branch": Experiment.PREF_BRANCH_DEFAULT,
+    "addon_name": "Addon Name",
+    "addon_experiment_id": "addon_experiment_id",
+    "addon_testing_url": "https://www.example.com/testing.xpi",
+    "addon_release_url": "https://www.example.com/release.xpi",
```

Comment:
Not a blocker but I think you should break this apart. Right now, if I'm not mistaken, this monster dict contains the union of all the possible data. Either something like this:

```python
VARIANTS_FORM_FIXTURE_BASE_DATA = {
    "population_percent": "10",
    "firefox_version": Experiment.VERSION_CHOICES[-1][0],
    "firefox_channel": Experiment.CHANNEL_NIGHTLY,
    "client_matching": "en-us only please",
    "locales": [],
    "countries": [],
    ...
}
VARIANTS_FORM_FIXTURE_ADDONS_DATA = {
    "addon_name": "Addon Name",
    "addon_experiment_id": "addon_experiment_id",
    "addon_testing_url": "https://www.example.com/testing.xpi",
    "addon_release_url": "https://www.example.com/release.xpi",
    ...
}
VARIANTS_FORM_FIXTURE_PREFS_DATA = {
    ...
}
```
...or...

reduce it down to the bare minimum that is common for all formsets and in each's `setUp` do something like:
```python
    def setUp(self):
        super().setUp()
        self.data = copy.copy(VARIANTS_FORM_FIXTURE_DATA)
        self.data.update({
             "specific": "to",
             "justthis": "Class",
             ...
        })
```

Also, since it's a "constant" (uppercase variable), it should sit at the top of the module. Either that, or even better, put it where it belongs; inside the first test case class that needs it.

Reply:
I don't think we can do this, because we parameterize the tests across all the form classes, the data needs to be a valid superset for all the forms.